### ISSUE TITLE
Badly formatted `@throws` doxygen throughout geometry

### ISSUE THREAD

**Issue by SeanCurtis-TRI** [RobotLocomotion/drake#9912]
There are numerous examples in the `geometry/` folder where the conditions for throwing exceptions are documented as:

```c++
/**
...
 @throws std::logic_error    1. If the `source_id` does _not_ map to a
                             registered source,
                             2. If the `parent_id` does _not_ map to a known
                             frame or does not belong to the source, or
                             3. a context has been allocated.  */
```

This renders badly.

![image](https://user-images.githubusercontent.com/22159799/47927219-e2146380-de7f-11e8-8a89-701d4cf797e7.png)

These need to be cleaned up so that they render reasonably.

(TBD Exactly what the "best" alternative is.)

**Comment by EricCousineau-TRI** [comment]
@jamiesnape may have some suggestions.

**Comment by sherm1** [comment]
The way you formatted in PR #9813 came out beautifully in doxygen:
![image](https://user-images.githubusercontent.com/4088016/47930232-8dc1b180-de88-11e8-81eb-de49481e3e09.png)

That was
```
   @throws std::logic_error
                            1. if the group name is invalid,
                            2. if the property name is invalid, or
                            3. if the property type is not that specified.
```

**Comment by jamiesnape** [comment]
You need a line break (and optional whitespace) before each list item. So in the original example you need a line break after std::logic_error.

This would work:
```
  @throws std::logic_error
                          1. If the `source_id` does _not_ map to a
                             registered source,
                          2. If the `parent_id` does _not_ map to a known
                             frame or does not belong to the source, or
                          3. a context has been allocated.
*/
```

**Comment by jamiesnape** [comment]
I should add though, that will render horribly in the Python documentation. reStructuredText requires a line break before a list, but that would separate the list from the `@throws` tag in Doxygen as well as in the Python documentation generation script.

**Comment by SeanCurtis-TRI** [comment]
That's the conclusion I'm coming to; the enumerated list just doesn't seem to be compatible with the multitude of media through which it is to be consumed.  Siiiigh.

**Comment by SeanCurtis-TRI** [comment]
Note: PR #10999 resolves this for `scene_graph.h`; the rest of geometry still needs to be cleaned up.

**Comment by tehbelinda** [comment]
Remainder addressed by #12156, let me know if there's anything I missed
